doc/cephfs/snapshots.rst

```diff
@@ -0,0 +1,81 @@
+================
+CephFS Snapshots
+================
+
+CephFS snapshots create an immutable view of the file system at the point
+in time they are taken. CephFS support snapshots which is managed in a 
+special hidden subdirectory named ``.snap`` .Snapshots are created using
+``mkdir`` inside this directory.
+
+Snapshots can be exposed with a different name by changing the following client configurations.
+
+- ``snapdirname`` which is a mount option for kernel clients
+- ``client_snapdir`` which is a mount option for ceph-fuse.
+
+Snapshot Creation
+==================
+
+CephFS snapshot feature is enabled by default on new file systems. To enable 
+it on existing file systems, use the command below.
+
+.. code-block:: bash
+    
+    $ ceph fs set <fs_name> allow_new_snaps true
+
+When snapshots are enabled, all directories in CephFS will have a special ``.snap``
+directory. (You may configure a different name with the client snapdir setting if 
+you wish.)
+To create a CephFS snapshot, create a subdirectory under ``.snap`` with a name of 
+your choice. 
+For example, to create a snapshot on directory ``/file1/``, invoke ``mkdir /file1/.snap/snapshot-name``
+
+.. code-block:: bash
+
+    $ touch file1
+    $ cd .snap
+    $ mkdir my_snapshot
+
+Using snapshot to recover data
+===============================
+
+Snapshots can also be used to recover some deleted files.
+
+- ``create a file1 and create snapshot snap1``
+.. code-block:: bash
```

Comment:
The line that begins with ``-`` and the line that begins with ``.. code-block:: bash`` must be separated by a blank line, or the docs build will fail. I force-pushed a change that made this alteration, but then your force push overwrote my force push. Let me know what you want to do in this connection.

Reply:
I missed you pushed change and it got over-wrote.I addressed that in new push

Reply:
When this PR has passed its build checks, I'm going to merge and backport it to S, R, and Q.

Reply:
> When this PR has passed its build checks, I'm going to merge and backport it to S, R, and Q.

No, this PR needs review from @ceph/cephfs too for further additions, if required

Reply:
I jumped the gun and merged this already. Your comment appeared in my browser only after I merged. But it's cool. We can always create another docs PR on top of the one that was raised here. I'll do a grammar pass sometime in the next twenty-four hours, for example. The upshot is this: docs PRs are not like code PRs. Docs PRs don't have to be perfect the first time. We can always alter them.

Reply:
@neesingh-rh, See https://github.com/ceph/ceph/pull/61249 for the continuation of this matter.